railties/lib/rails/application/reloader.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require "active_support/reloader"
+
+module Rails
+  class Application
+    class Reloader < ::ActiveSupport::Reloader
+      class_attribute :registered_reloaders, default: {}
+
+      def self.register(name, &block)
```

Comment:
We need to document the methods we want people to use. I think they are `register` and `fetch`. And `# :nodoc:` the methods we don't want people to use.